File: integrations/src/main/java/com/nivasafinance/integrations/framework/core/entity/ThirdPartyServiceConfig.java
package com.nivasafinance.integrations.framework.core.entity;

import com.nivasafinance.common.audit.AuditableEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "n_third_party_service_config")
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class ThirdPartyServiceConfig extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "service")
    private String service;

    @Column(name = "primary_config" , nullable = false)
    private Long primaryConfigId;

    @Column(name = "fallback_config")
    private Long fallbackConfigId;

    @Column(name = "retry_count")
    private Integer retryCount;

    @Column(name = "is_primary")
    private Boolean isPrimary;

    @Column(name = "is_active")
    private Boolean isActive;
}

File: integrations/src/main/kotlin/framework/core/entity/ThirdPartyServiceConfig.kt
package framework.core.entity

import audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID
import org.javers.core.metamodel.annotation.Entity as JaversEntity

@Table(name = "f_third_party_service_config")
@Entity
@JaversEntity
data class ThirdPartyServiceConfig(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID,

    @Column(name = "name")
    val name: String,

    @Column(name = "service")
    val service: String,

    @Column(name = "primary_config_key")
    val primaryConfigKey: UUID,

    @Column(name = "fallback_config_key")
    val fallbackConfigKey: UUID? = null,

    @Column(name = "retry_count")
    val retryCount: Int,

    @Column(name = "is_primary")
    val isPrimary: Boolean,

    @Column(name = "is_active")
    val isActive: Boolean,
) : AuditableEntity()
